src/System.Text.Json/src/System/Text/Json/JsonReaderState.cs

```diff
@@ -22,6 +22,7 @@ public struct JsonReaderState
         internal int _maxDepth;
         internal bool _inObject;
         internal bool _isNotPrimitive;
+        internal char _numberFormat;
```

Comment:
I am trying to see if we need to persist number format as part of the JsonReaderState.

It looks like this field is transient and only relevant when a particular (numeric) JSON element has been read. It doesn't affect subsequent reads and not needed when a new Utf8JsonReader instance is started to re-start reading from a partially read state.

Unless there is a scenario that is blocked if we remove it, I think we shouldn't add the field to the state.